scripts/zones/Silver_Sea_route_to_Nashmau/Zone.lua

```diff
@@ -7,26 +7,41 @@ local ID = require("scripts/zones/Silver_Sea_route_to_Nashmau/IDs")
 -----------------------------------
 
 function onInitialize(zone)
-end;
+end
 
 function onZoneIn(player,prevZone)
-    local cs = -1;
-
-    return cs;
-end;
+    local cs = -1
+    -- Gets current vanadiel time and compares it with arrival time to start a timer to show a message before it docks.
+    local vHour = VanadielHour()
+    local vMin  = VanadielMinute()
+    local dockHour
+    local dockMin = 40
+    if vHour < 5 then
+        dockHour = 4
+    elseif vHour < 13 then
+        dockHour = 12
+    elseif vHour < 21 then
+        dockHour = 20
+    end
+    local earthSecsAtZoneIn = math.floor(2.4 * ((vHour * 60) + vMin))
+    local earthSecsAtDocking = math.floor(2.4 * ((dockHour * 60) + dockMin))
```

Comment:
will do. I just added the var in case it was seen as a magic number lol. Will be happy to remove it.

I'll play around with your other suggestion :cat:

Reply:
@ibm2431  im think i wanna keep the minute declaration so it can be easily understood when reading the formula.

I am not doing a static docktime as the reason why i did this is because im taking in consideration the time the player zones into the zone to start the timer, if for example they are running on a potatoe and the loading area needs more time time before the player gets loaded into the zone then if using a static time, they might never see the message. Or also if they log out or disconnect and reconnect while still on the ship then the timer can take in consideration the time they just zoned back into the ship.

Sorry if it sounds complicated but we can take it on discord to try to explain myself a bit better